client_internals_it_test.go

```diff
@@ -542,3 +543,176 @@ func ensureClient(config hz.Config) *hz.Client {
+			muStatus: OPTIONAL,
+			ca:       "testdata/OpenSSL/rootCA.crt",
+			cert:     "",
+			key:      "",
+			wantErr:  false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			skip.IfNot(t, "oss")
+			if tc.wantErr {
+				require.Error(t, sslTestUtil(t, tc.muStatus, tc.ca, tc.cert, tc.key))
+			} else {
+				require.NoError(t, sslTestUtil(t, tc.muStatus, tc.ca, tc.cert, tc.key))
+			}
+		})
+	}
+}
+
+func sslTestUtil(t *testing.T, mu MutualAuthenticationStatus, ca, cert, key string) error {
+	ctx := context.Background()
+	port := it.NextPort()
+	clsConfig := xmlSSLMutualAuthenticationConfig(t.Name(), port, mu)
+	tc := it.StartNewClusterWithConfig(1, clsConfig, port)
+	defer tc.Shutdown()
+	clientCfg := tc.DefaultConfig()
+	if err := clientCfg.Cluster.Network.SSL.SetCAPath(ca); err != nil {
+		return err
+	}
+	// unset certificate and key should not be added
+	if len(cert) != 0 && len(key) != 0 {
+		if err := clientCfg.Cluster.Network.SSL.AddClientCertAndKeyPath(cert, key); err != nil {
+			return err
+		}
+	}
+	client, err := hz.StartNewClientWithConfig(ctx, clientCfg)
+	if err != nil {
+		return err
+	}
+	if err = client.Shutdown(ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+func xmlSSLMutualAuthenticationConfig(clusterName string, port int, mu MutualAuthenticationStatus) string {
+	return fmt.Sprintf(`
+		<hazelcast xmlns="http://www.hazelcast.com/schema/config"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://www.hazelcast.com/schema/config
+           http://www.hazelcast.com/schema/config/hazelcast-config-4.0.xsd">
+			<cluster-name>%s</cluster-name>
+			<network>
+				<port>%d</port>
+				<ssl enabled="true">
+					<factory-class-name>
+						com.hazelcast.nio.ssl.ClasspathSSLContextFactory
+					</factory-class-name>
+					<properties>
+						<property name="keyStore">testdata/OpenSSL/ca.keystore</property>
+						<property name="keyStorePassword">123456</property>
+						<property name="trustStore">testdata/OpenSSL/ca.truststore</property>
+						<property name="trustStorePassword">123456</property>
+						<property name="trustManagerAlgorithm">SunX509</property>
+						<property name="javax.net.ssl.mutualAuthentication">%s</property>
+						<property name="keyManagerAlgorithm">SunX509</property>
+						<property name="protocol">TLSv1.2</property>
+					</properties>
+				</ssl>
+			</network>
+			<map name="test-map">
+				<map-store enabled="true">
+					<class-name>com.hazelcast.client.test.SampleMapStore</class-name>
+				</map-store>
+			</map>
+			<serialization>
+				<data-serializable-factories>
+					<data-serializable-factory factory-id="66">com.hazelcast.client.test.IdentifiedFactory</data-serializable-factory>
+					<data-serializable-factory factory-id="666">com.hazelcast.client.test.IdentifiedDataSerializableFactory</data-serializable-factory>
+				</data-serializable-factories>
+			</serialization>
```

Comment:
These shouldb't be required for the SSL tests.

Reply:
Removed, [c6481ec](https://github.com/hazelcast/hazelcast-go-client/pull/859/commits/c6481ec04328cf044ab3bec8e31bd712bd663bd2).